### PR TITLE
[BFT] Improve network message validation

### PR DESCRIPTION
**Pull request by peterargue**
Improve handling for several network message fields:
* Explicitly validate `msg.ChannelID` matches the `Topic` for pubsub messages
* Generate the `msg.EventID` and `msg.Type`

### CODE REVIEW

File: network/validator/pubsub/topic_validator.go [hunk 1]
```diff
@@ -88,37 +88,59 @@ func TopicValidator(log zerolog.Logger, c network.Codec, slashingViolationsConsu
...
+
+		channel := channels.Channel(msg.ChannelID)
+		if channel != actualChannel {
+			log.Warn().
+				Str("actual_channel", actualChannel.String()).
+				Bool(logging.KeySuspicious, true).
+				Msg("channel id in message does not match pubsub topic")
+			return pubsub.ValidationReject
+		}
+
```
[the remark this answers is not in the document]
Comment (yhassanzadeh13) [reply network/validator/pubsub/topic_validator.go:125]: Please also [deprecated](https://github.com/onflow/flow-go/blob/fdcb6b376d6ff90f4dc6fe89b62ec2be480e6f31/network/message/message.proto#L12) `EventID`, `Type` and `ChannelID` and [rebuild](https://github.com/onflow/flow-go/blob/fdcb6b376d6ff90f4dc6fe89b62ec2be480e6f31/network/message/message.proto#L5) `message.proto` file.

[the remark this answers is not in the document]
Comment (peterargue) [reply network/validator/pubsub/topic_validator.go:125]: 0428da9

**Review by vishalchangrani — APPROVED** [review]
LGTM ❤️ - nice work

**Comment by peterargue** [comment]
bors merge

**Pull request merged**